core/src/main/java/discord4j/core/event/domain/InviteCreateEvent.java

```diff
@@ -0,0 +1,183 @@
+import discord4j.core.object.entity.User;
+import discord4j.core.object.util.Snowflake;
+import reactor.core.publisher.Mono;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Optional;
+
+/**
+ * Dispatched when a channel invite is created.
+ * <p>
+ * This event is dispatched by Discord.
+ *
+ * @see <a href="https://discordapp.com/developers/docs/topics/gateway#invite-create">Invite Create</a>
+ */
+public class InviteCreateEvent extends Event {
+
+    private final long guildId;
+    private final long channelId;
+    private final String code;
+    private final User inviter;
+    private final Instant createdAt;
+    private final int uses;
+    private final int maxUses;
+    private final int maxAge;
+    private final boolean temporary;
+
+    public InviteCreateEvent(DiscordClient client, long guildId, long channelId, String code, User inviter, Instant createdAt, int uses, int maxUses, int maxAge, boolean temporary) {
+        super(client);
+        this.guildId = guildId;
+        this.channelId = channelId;
+        this.code = code;
+        this.inviter = inviter;
+        this.createdAt = createdAt;
+        this.uses = uses;
+        this.maxAge = maxAge;
+        this.maxUses = maxUses;
+        this.temporary = temporary;
+    }
+
+    /**
+     * Gets the {@link Snowflake} ID of the guild that had a invite created in this event.
+     *
+     * @return The ID of the guild involved.
+     */
+    public Snowflake getGuildId() {
+        return Snowflake.of(guildId);
+    }
+
+    /**
+     * Requests to retrieve the {@link Guild} that had a invite created in this event.
+     *
+     * @return A {@link Mono} where, upon successful completion, emits the {@link Guild} involved in the event.
+     * If an error is received, it is emitted through the {@code Mono}.
+     */
+    public Mono<Guild> getGuild() {
+        return getClient().getGuildById(getGuildId());
+    }
+
+    /**
+     * Gets the {@link Snowflake} ID of the channel invite belongs to.
+     *
+     * @return The ID of the channel involved.
+     */
+    public Snowflake getChannelId() {
+        return Snowflake.of(channelId);
+    }
+
+    /**
+     * Gets the invite code (unique ID).
+     *
+     * @return The invite code (unique ID).
+     */
+    public final String getCode() {
+        return code;
+    }
+
+    /**
+     * Gets the ID of the target user this invite is associated to, if present.
+     *
+     * @return The ID of the target user this invite is associated to, if present.
+     */
+    public final Snowflake getInviterId() {
+        return inviter.getId();
+    }
+
+    /**
+     * Requests to retrieve the user who created the invite.
+     *
+     * @return A {@link Mono} where, upon successful completion, emits the {@link User user} who created the invite. If
+     * an error is received, it is emitted through the {@code Mono}.
+     */
+    public Mono<User> getInviter() {
```

Comment:
Why don't you return `inviter` ?

Reply:
I just saw your message about user having all fields to null, that could be a bug...

Reply:
Well its a partial user i dont know if only give the User instance or "request" using the ID.

Reply:
You don't even get the ID `UserBean{id=0, username='null', discriminator='null', avatar='null', isBot=false}`

Reply:
With Mono you say?
i test before the change to Mono and the event return.
`InviteCreateEvent{code='HvArhW', guildId=545819046294192128, channelId=546495205189222411, inviter=User{data=UserBean{id=216316781483130880, username='Doc', discriminator='4444', avatar='a_891b451892dfa757d392c53377f526a1', isBot=false}}, uses=0, maxUses=0, maxAge=86400, temporary=false, createdAt='2020-01-18T20:56:58.478399Z'}`

Reply:
While documentation says it is a partial user, I don't really know which field is missing ? You even get isBot

Reply:
i dont test if BOT make the invite work, but the system looks... working.

Reply:
I was talking about isBot because the documentation does not say that this field is provided in the partial user object

Reply:
yes i have the same question but in testing i see the values.
Test delete/create using BOT, value working.
![image](https://user-images.githubusercontent.com/3602279/72684470-6bcd4280-3abf-11ea-85e6-b2282dc57ce1.png)

Reply:
In fact, DAPI does not send if the user is a bot, this is assumed as false if null by Discord4J. This is why it is a partial user.
I'm sorry, you can do what you did before by doing a REST / cache request to retrieve the full user.